packages/image-block/src/components/Image.tsx

```diff
@@ -27,14 +28,14 @@ export const ImageComponent = ({
     isEditing,
     appBridge,
     isAssetViewerEnabled,
-}: ImageProps & { isAssetViewerEnabled: boolean }) => {
+    containerWidth,
+}: ImageProps & { isAssetViewerEnabled: boolean; containerWidth: number }) => {
     const { open } = useAssetViewer(appBridge);
     const link = blockSettings?.hasLink && blockSettings?.linkObject?.link && blockSettings?.linkObject;
     const { isFocused, focusProps } = useFocusRing();
 
     const devicePixelRatio = Math.max(1, window?.devicePixelRatio ?? 1);
-    const imageWidthToRequest = Math.min(800 * devicePixelRatio, image.width);
-
+    const imageWidthToRequest = Math.min(containerWidth * devicePixelRatio, image.width);
```

Comment:
Just out of curiosity do you know if there would be a benefit from having a list of image sizes, and then picking the lowest size that fits the container? I'm wondering if this would help somehow with caching the images. For example:
We could have sizes [240, 360, 480, 720 , 1080, 1440] and. if the container is 800px it will request the 1080 version. Not sure if this would help

Reply:
The reason why I think this might be something good to add is also because at the minute if the block changes size the containerWidth state does not update. Especially when it comes to sections I can imagine a situation where the user has an image block as one of 3 blocks on a row. Since the user could load the page at a certain width, resizing the page could cause the image block to wrap and then actually expand from it's original size, since it will now be on a row by itself and the image will be too low quality.

We could run a resizeObserver to see which size in the array matches best. Since it would be using the array value it would have better performance than using the raw width as it would only trigger a state change when a new array size is reached (eg: container.width: 300 => 400 changes the image width from '360' to '480', and only modifies the state once. 300 => 350 causes no state changes to occur). 
Minor edge case though so it could be a separate pr.

Reply:
I like this approach, seems like a sweet spot between using `srcset` and hardcoding one value. Small concern of mine is how much would it hurt the performance to attach a resize observer for each image block, but will need to discover this.

For this I would prefer a separate PR and ticket.

Reply:
Sure, sounds good 👍